internal/name_persister.go

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package internal
+
+import "context"
+
+type NamePersister interface {
+	Set(name string, value string)
+	Get(name string) (string, bool)
+	Reset(name string)
+	PersistenceInfo() map[string]string
+}
+
+func PersisterFromContext(ctx context.Context) NamePersister {
+	return ctx.Value("persister").(NamePersister)
+}
+
+func SetContext(ctx context.Context, persister NamePersister) context.Context {
```

Comment:
Would something like `ContextWithPersister` work better as the name of the function?

Reply:
Thanks both, fixed at https://github.com/hazelcast/hazelcast-commandline-client/pull/61/commits/1cf0d5783c02506d0201a88c55212ee671b61097